generate_helm_release.sh

```diff
@@ -51,7 +51,7 @@ main() {
         git stash
         remote=$(git remote -v | grep "${ORG}/${PROJECT}" | awk '{print $1;exit}')
         git fetch "${remote}"
-        git checkout -b "$version"
+        git checkout -B "$version"
```

Comment:
@kkourt Do you recall the circumstances in which you found this necessary?

I was looking into why we've been hitting problems creating helm charts for the past several months and looking at this change I think it's incorrect. The assumption of this conditional branch is that there's a copy of the repository checked out, and we must switch over to the target release version. Therefore `$version` *must* exist in the upstream repository, because this is the version that this script is preparing Helm charts for. If the target version isn't already tagged then we cannot create Helm charts for it.

Reply:
Fix: https://github.com/cilium/charts/pull/166

Reply:
I don't recall exactly. I think I ended up in a situation where the local branch `$version` existed, and the script would fail because `-b` fails if the local branch already exists.

But the `-B` choise was wrong as you said:
> If -B is given, <new-branch> is created if it doesn’t exist; otherwise, it is reset.

My guess is that I've misssed the "other, it is reset" part or did not really understood what it did. 
Thanks for the fix.